### PR TITLE
querier: Added a tests with real data after promql processing.

### PR DESCRIPTION
**Pull request by krasi-georgiev**

### CODE REVIEW

File: pkg/query/querier_test.go [hunk 1]
```diff
@@ -576,12 +675,93 @@ func testSelectResponse(t *testing.T, expected []series, res storage.SeriesSet)
...
+	return series
+}
+
 type mockedQueryable struct {
-	q storage.Querier
+	Creator func(int64, int64) storage.Querier
+	querier storage.Querier
+}
+
+// Querier creates a queirier with the provided min and maxt.
```
Comment (bwplotka) [inline pkg/query/querier_test.go:745]:
```suggestion
// Querier creates a querier with the provided min and max time.
```

File: pkg/query/querier_test.go [hunk 2]
```diff
@@ -576,12 +675,93 @@ func testSelectResponse(t *testing.T, expected []series, res storage.SeriesSet)
...
+// The promq engine sets mint and it is calculated based on the default lookback delta.
```
Comment (bwplotka) [inline pkg/query/querier_test.go:746]:
```suggestion
// The promQL engine sets mint and it is calculated based on the default lookback delta.
```

File: pkg/query/querier_test.go [hunk 3]
```diff
@@ -292,6 +294,103 @@ func (s series) Iterator() chunkenc.Iterator {
...
+// TestQuerier_Select_After_promql tests expected results with and without deduplication after passing all data to promql.
+// To test with real data:
+// Collect the expected results from Prometheus or Thanos through "/api/v1/query_range" and save to a file.
+// Collect raw data to be used for local storage:
+// 	scripts/insecure_grpcurl_series.sh querierGrpcIP:port '[{"name":"type","value":"current"},{"name":"_id","value":"xxx"}]' 1597823000000 1597824600000 > localStorage.json
+// 	Remove all white space from the file and put each series in a new line.
+// 	When collecting the raw data mint should be Prometheus query time minus the default look back delta(default is 5min or 300000ms)
+// 	For example if the Prometheus query mint is 1597823700000 the grpccurl query mint should be 1597823400000.
+//  This is because when promql displays data for a given range it looks back 5min before the requested time window.
+func TestQuerier_Select_After_promql(t *testing.T) {
```
Comment (bwplotka) [inline pkg/query/querier_test.go:306]:
```suggestion
func TestQuerier_Select_AfterPromQL(t *testing.T) {
```

File: pkg/query/querier_test.go [hunk 4]
```diff
@@ -292,6 +294,103 @@ func (s series) Iterator() chunkenc.Iterator {
 	return newMockedSeriesIterator(s.samples)
 }
 
+// TestQuerier_Select_After_promql tests expected results with and without deduplication after passing all data to promql.
```
Comment (bwplotka) [inline pkg/query/querier_test.go:297]:
```suggestion
// TestQuerier_Select_AfterPromQL tests expected results with and without deduplication after passing all data to promql.
```

**Review by bwplotka — APPROVED** [review]
Nice, thanks! Some tiny comments only.

File: pkg/query/querier_test.go (hunk 2 above)
Comment (krasi-georgiev) [reply pkg/query/querier_test.go]: updated

File: pkg/query/querier_test.go (hunk 4 above)
Comment (krasi-georgiev) [reply pkg/query/querier_test.go]: updated

File: pkg/query/querier_test.go (hunk 3 above)
Comment (krasi-georgiev) [reply pkg/query/querier_test.go]: updated

File: pkg/query/querier_test.go (hunk 1 above)
Comment (krasi-georgiev) [reply pkg/query/querier_test.go]: updated

**Review by kakkoyun — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**